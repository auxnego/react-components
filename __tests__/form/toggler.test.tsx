import React from "react";
import {create} from "react-test-renderer";
import { render, act } from "@testing-library/react";
import Toggler from "../../form/toggler";

it("normal name=basic", () => {

  expect(create(
    <Toggler name="basic" />
  ).toJSON()).toMatchSnapshot();

});

it("normal name=basic and initialStatus=true", () => {

  expect(create(
    <Toggler name="basic" initialStatus={true} />
  ).toJSON()).toMatchSnapshot();

});

it("normal name=basic and statusText Encendido/Apagado", () => {

  expect(create(
    <Toggler name="basic" statusText={{on:"Encendido", off:"Apagado"}}/>
  ).toJSON()).toMatchSnapshot();

});

it("normal name=basic and onValue=true and offValue=false", () => {

  expect(create(
    <Toggler name="basic" onValue="true" offValue="false" />
  ).toJSON()).toMatchSnapshot();

});

it("change state with click", () => {

  let { container }  = render(<Toggler name="basic"/>);

  act(() => {
   let tree =  container.querySelector(".toggler__root") as HTMLElement;
   tree.click();
  });

  expect(container).toMatchSnapshot();

});
