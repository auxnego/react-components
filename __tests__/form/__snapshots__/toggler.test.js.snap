// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`change state with click 1`] = `
<div>
  <div
    class="toggler__root toggler__on"
  >
    <div
      class="toggler__input"
    >
      <input
        id="basic"
        name="basic"
        type="hidden"
        value="1"
      />
      <div
        class="toggler__on toggler__on--actived"
      />
      <div
        class="toggler__off"
      />
    </div>
  </div>
</div>
`;

exports[`normal name=basic 1`] = `
<div
  className="toggler__root"
  onClick={[Function]}
>
  <div
    className="toggler__input"
  >
    <input
      id="basic"
      name="basic"
      type="hidden"
      value="0"
    />
    <div
      className="toggler__on"
    />
    <div
      className="toggler__off toggler__off--actived"
    />
  </div>
</div>
`;

exports[`normal name=basic and initialStatus=true 1`] = `
<div
  className="toggler__root toggler__on"
  onClick={[Function]}
>
  <div
    className="toggler__input"
  >
    <input
      id="basic"
      name="basic"
      type="hidden"
      value="1"
    />
    <div
      className="toggler__on toggler__on--actived"
    />
    <div
      className="toggler__off"
    />
  </div>
</div>
`;

exports[`normal name=basic and onValue=true and offValue=false 1`] = `
<div
  className="toggler__root"
  onClick={[Function]}
>
  <div
    className="toggler__input"
  >
    <input
      id="basic"
      name="basic"
      type="hidden"
      value="false"
    />
    <div
      className="toggler__on"
    />
    <div
      className="toggler__off toggler__off--actived"
    />
  </div>
</div>
`;

exports[`normal name=basic and statusText Encendido/Apagado 1`] = `
<div
  className="toggler__root"
  onClick={[Function]}
>
  <div
    className="toggler__input"
  >
    <input
      id="basic"
      name="basic"
      type="hidden"
      value="0"
    />
    <div
      className="toggler__on"
    />
    <div
      className="toggler__off toggler__off--actived"
    />
  </div>
  <label
    className="toggler__label"
    htmlFor="basic"
  >
    Apagado
  </label>
</div>
`;
